maven-model-builder/src/main/java/org/apache/maven/model/superpom/DefaultSuperPomProvider.java

```diff
@@ -52,8 +53,8 @@ public DefaultSuperPomProvider(ModelProcessor modelProcessor) {
 
     @Override
     public Model getSuperModel(String version) {
-        if (superModel == null) {
-            String resource = "/org/apache/maven/model/pom-" + version + ".xml";
+        return SUPER_MODELS.computeIfAbsent(version, v -> {
```

Comment:
Does this mean that for example maven 4.0.1 could return super POM of Maven 3.8.6 (ie. newer could have all super POMs of older Mavens?)

Reply:
It could, though the only place it's called is with `4.0.0` afaik, and that's since ages.  I found the code does not reflect the javadoc and that the `version` parameter is not used, so I fixed it.
If that's not intended, then we should remove the parameter completely.
The version was originally hardcoded, see https://github.com/apache/maven/commit/dd635ca490bae271442d7b69eafdf595a627cafd
The commit added a version parameter, but it's only used the first time and subsequent calls return the cached version.

As for having multiple superpoms, I think the problem is that the version has never been updated when the pom changed.  So not sure what the future will be for the supermom...